### PR TITLE
Scan: implement request index page top part

### LINKED ISSUE #2430: Revamp GuanYu - Implement Request Info (Request Index)

### LINKED ISSUE #2431: Revamp GuanYu - Implement Calldata, Result, External Data (Request Info)

### LINKED ISSUE #2432: Revamp GuanYu - Implement Data Report List (Request Index)

### PR DESCRIPTION
**Pull request by evilpeach**
Fixed: #2430 
Fixed: #2431 
Fixed: #2432 

## Implementation details
- Request Info
- reported validator chart
- Calldata table
- Result table
- Proof
- External Data
- Support Mobile & Desktop

## Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [x] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [x] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any and all appropriate unit/integration tests
- [x] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [x] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

**Comment by pzshine** [comment]
image

I just checked on mobile screen and found only one point is about buttons.

**Comment by evilpeach** [comment]
> image
> 
> I just checked on mobile screen and found only one point is about buttons.

Do we change in this PR or make another PR for changing all button styles?

File: scan/src/pages/RequestIndexPage.re [hunk 1]
```diff
@@ -1,384 +1,592 @@
...
+  <Section>
+    <div className=CssHelper.container>
+      <Row.Grid marginBottom=40 marginBottomSm=16>
+        <Col.Grid>
+          <Heading value="Data Request" size=Heading.H4 marginBottom=40 marginBottomSm=24 />
+          {switch (requestSub) {
+           | Data({id}) =>
+             <div className={CssHelper.flexBox()}>
+               <TypeID.Request id position=TypeID.Title />
+             </div>
```
Comment (pzshine) [inline scan/src/pages/RequestIndexPage.re]: No need to use flexBox

File: scan/src/pages/RequestIndexPage.re [hunk 2]
```diff
@@ -1,384 +1,592 @@
...
+            </Row.Grid>
+            <Row.Grid>
+              <Col.Grid col=Col.Six mbSm=24>
+                <Heading value="TX Hash" size=Heading.H5 />
+                <VSpacing size=Spacing.sm />
+                {switch (requestSub) {
+                 | Data({transaction: {hash}}) =>
+                   <div className={CssHelper.flexBox()}>
+                     <TxLink txHash=hash width={isMobile ? 260 : 360} />
+                   </div>
```
Comment (pzshine) [inline scan/src/pages/RequestIndexPage.re]: No need to use flexBox

File: scan/src/pages/RequestIndexPage.re [hunk 3]
```diff
@@ -1,384 +1,592 @@
...
+                     <Text
+                       block=true
+                       value={
+                         (gasFee |> Coin.getBandAmountFromCoins |> Format.fPretty(~digits=2))
+                         ++ " BAND"
+                       }
+                       size=Text.Lg
+                       color=Colors.gray7
+                     />
+                   </div>
```
Comment (pzshine) [inline scan/src/pages/RequestIndexPage.re]: No need to use flexBox

File: scan/src/pages/RequestIndexPage.re [hunk 4]
```diff
@@ -1,384 +1,592 @@
...
+                <VSpacing size={`px(8)} />
+                {switch (requestSub) {
+                 | Data({resolveStatus}) =>
+                   <div className={CssHelper.flexBox()}>
+                     <RequestStatus resolveStatus display=RequestStatus.Full />
                      <HSpacing size=Spacing.sm />
-                     <img src=Images.fail className={Styles.logo(20, 0)} />
-                   </>
-                 | _ => React.null
+                   </div>
```
Comment (pzshine) [inline scan/src/pages/RequestIndexPage.re]: No need to use flexBox

**Review by pzshine — CHANGES_REQUESTED** [review]

**Review by pzshine — APPROVED** [review]

**Pull request merged**